Update project and code / fix bugs (More to come)

leonard-colin opened the pull request:
I updated some libraries, fixed some bugs and I mainly added Pipenv to manage the environment and make the install easier.

leonard-colin commented:
Hey there ! 

I really do like what you created with this project.
I just installed it in my home, like you with a picture frame.

I spent some time actually to make it work with updated libraries, and also know what packages should be installed.
Therefore I added a pipenv environment to make it easier to install.
I also plan to write a small README file.

Your project deserves to be maintained ! :)

So if that's ok with you, I'd like to make some PR from time to time.

Thank you for your work !

ALinkToTao commented:
> Hey there !
> 
> I really do like what you created with this project. I just installed it in my home, like you with a picture frame.
> 
> I spent some time actually to make it work with updated libraries, and also know what packages should be installed. Therefore I added a pipenv environment to make it easier to install. I also plan to write a small README file.
> 
> Your project deserves to be maintained ! :)
> 
> So if that's ok with you, I'd like to make some PR from time to time.
> 
> Thank you for your work !

@leonard-colin, is there any way that I could pay someone to build one of these for me?  I'd pay for the supplies plus a fair hourly rate (you tell me what that is) to program and build it.

I've been wanting one of these for a really long time, but I don't have the time nor do I have good skills with programming to get it working.

I agree with you 100%, this project needs to be maintained.

leonard-colin commented:
> > Hey there !
> > I really do like what you created with this project. I just installed it in my home, like you with a picture frame.
> > I spent some time actually to make it work with updated libraries, and also know what packages should be installed. Therefore I added a pipenv environment to make it easier to install. I also plan to write a small README file.
> > Your project deserves to be maintained ! :)
> > So if that's ok with you, I'd like to make some PR from time to time.
> > Thank you for your work !
> 
> @leonard-colin, is there any way that I could pay someone to build one of these for me? I'd pay for the supplies plus a fair hourly rate (you tell me what that is) to program and build it.
> 
> I've been wanting one of these for a really long time, but I don't have the time nor do I have good skills with programming to get it working.
> 
> I agree with you 100%, this project needs to be maintained.

Hello @ALinkToTao 
Of course, with pleasure !
Let me reach out to you in the next few days and we'll try to do something :)

depuntism commented:
Hello @leonard-colin,

Is there any way to autorestart the raspberry pi at a specific time? Mine always have the code frozen after some days of uses and I still haven't been able to figure out why it reacts like that... It doesn't seem to overheat or be overwhelmed by lack of ram or such. I have to manually restart the raspberry pi for the code to work again.

I tried crontab but it didn't seemed to work...

I'm out of option but I don't want to xcrap this project because this is a really good one and it was working really well.

depuntism commented:
By the way, what is the recommended pi os version for it to work?

leonard-colin commented:
> Hello @leonard-colin,
> 
> Is there any way to autorestart the raspberry pi at a specific time? Mine always have the code frozen after some days of uses and I still haven't been able to figure out why it reacts like that... It doesn't seem to overheat or be overwhelmed by lack of ram or such. I have to manually restart the raspberry pi for the code to work again.
> 
> I tried crontab but it didn't seemed to work...
> 
> I'm out of option but I don't want to xcrap this project because this is a really good one and it was working really well.

Hello @depuntism,
Indeed, I myself need to restart the raspi manually, as the code freezes sometimes.
It was actually my next fix to come. But I need to get some logs and understand where it freezes.
Regarding the raspi autorestart, I didn't try yet.
I need to find some time to do all this, but I will !

Also, @ALinkToTao, sorry for not reaching you back but I don't have enough time to spare on this at the moment.

leonard-colin commented:
> By the way, what is the recommended pi os version for it to work?

I think the latest stable is good. You can even take the one without desktop

depuntism commented:
Hello again @leonard-colin, I think I solved the problem. For some reason, I was no more able to access SSH from my PC and it was no more updating at this exact moment. I added this command on the /etc/modprobe.d/8192cu.conf file and until now, it seems to be working flawlessly: options 8192cu rtw_power_mgnt=0 rtw_enusbss=0

By the way, is the saved.txt option still in use with your fork because I cannot see the graph anymore which I liked on the original project.

leonard-colin commented:
> Hello again @leonard-colin, I think I solved the problem. For some reason, I was no more able to access SSH from my PC and it was no more updating at this exact moment. I added this command on the /etc/modprobe.d/8192cu.conf file and until now, it seems to be working flawlessly: options 8192cu rtw_power_mgnt=0 rtw_enusbss=0
> 
> By the way, is the saved.txt option still in use with your fork because I cannot see the graph anymore which I liked on the original project.

Hello @depuntism,
Nice catch !
Not sure it's the same problem I have but I'll try it out.
Oh yes sorry for the graph. I have to make a README at this point I think..
I added `argparse` to add a  `--show-graph` argument when you run the project. The main reason is I never look at the graphs and the news are always cut... So I made an option to remove the graphs and extend de news.

So, if you add the show-graph argument, you should see the graph. But tell me if the display remains correct then. Not sure if I made it look nice in both cases (with and without the graphs)

depuntism commented:
Hey @leonard-colin,

After three or four days, even with the fixes I thought would resolve the no more updating problem, it happened again... No more updates from the script and even the raspberry pi completely went bonkers and I wasn't able to do anything on it even after restarting it.
What I've done is that instead of having the main script running all the time, I've run it every 15 minutes with a cronjob and, if the raspberry pi shuts down due to a power outage, I've set up a script that runs on reboot and recreates the cronjob if it can't find it in the crontab.
Seems to be working so far and I think it's much less of a memory hog working this way.

Let's wait and see if it works well on the long run.

I abandonned the graph because I wasn't able to make it work and I understand your point about the news being cut off.

Also, be sure to tell on your readme how to add their language on the raspberry pi itself via the raspi-config because I had to add it manually to have the days in french.

Kind regards,

leonard-colin commented:
Hi @depuntism,

Nice to hear you may have found a solution 
Thanks for the detailed insight. I'll try this solution as well.

And of course you can fork it, I think this project is made to be forked and improved by everyone that have the time and motivation to make it better and better.

I'll be interested to see what you've done with it also.

Let's keep in touch then!

Have a great day
